### ISSUE TITLE
page size of "all"

### ISSUE THREAD

**Issue by ajax-zoom** [Mottie/tablesorter#1366]
v2.26.0, a page size of "all" instead of a specific number will be saved and used on reload... forces to have "all" as an option(value) in select which is wrong for the following reasons:

1. If you have 100.000 entries in the database it will surely break the browser / ajax request are not completed. This is already the case for 3000+ entries. So even if we set a limit for "all" on server side, this is not "All". 
2. `maxOptionSize` does not make sense any more
3. Inconsistent behavior - when you have 10 entries in the database, the script selects "all" by actually sending a request of showing 10 rows. If you have 9 entries in the database, `parsePageSize` returns 10 as selected. 

What is the sense of forcing "all"? In case there is a reason behind it which we just not understand, please implement it optional. The only changes needed are inside `parsePageSize` and `updatePageDisplay` methods + an option for forcing "all". 

Also, AJAX requests might take some time... if we would like to implement a customer loader or maybe something else on start, these couple lines would be perfect in `getAjax` method (`onBeforeGetAjaxFirst` and `onBeforeGetAjax` are options ):

```
if (p.ajaxCounter == 0 && typeof p.onBeforeGetAjaxFirst === 'function')  {
	p.onBeforeGetAjaxFirst();
}

if (p.ajaxCounter > 0 && typeof p.onBeforeGetAjax === 'function')  {
	p.onBeforeGetAjax();
}
```

**Comment by ajax-zoom** [comment]
@Mottie: #1364 does not really solve the issue completly (consider filters etc.). In our opinion parsePageSize method should not return or save "all" if there is no "all" in the pager as an option. "all" should not be sent to server, at least not if configuration does not allow it, so the easiest way is to autodetect "all" in pager or make sending all optionally. We are using a customized version of pager which does work as expected but would like to stick to the project in the long run...

Yes, customAjaxUrl could be abused to trigger a function before ajax request, we did not see it / it was not obvious :-)

Thank you.

**Comment by Mottie** [comment]
Hi @ajax-zoom!

This should now be fixed in v2.28.6. Please verify this on your end.

**Comment by Mottie** [comment]
I'm guessing this issue has been resolved, so I'm going to close it. If you continue to have problems, please feel free to continue the discussion in this thread.
